utils/S3UtilsMongoClient.js

```diff
@@ -39,72 +41,61 @@ class S3UtilsMongoClient extends MongoClientInterface {
-            res => {
-                const { data, error } = this._processEntryData(bucketName, bucketInfo, res, isTransient, locationConfig);
-
-                if (error) {
-                    log.error('Failed to process entry data', {
-                        method: 'getObjectMDStats',
-                        entry: res,
-                        error,
-                    });
-                }
-
-                let targetCount;
-                let targetData;
-                if (res._id.indexOf('\0') !== -1) {
-                    // versioned item
-                    targetCount = 'versionCount';
-                    targetData = 'versionData';
-
-                    if (res.value.replicationInfo.backends.length > 0
-                        && this._isReplicationEntryStalled(res, cmpDate)) {
-                        stalledCount++;
-                    }
-                } else if (!!res.value.versionId && !res.value.isNull) {
-                    // master version
-                    // includes current objects in versioned bucket and
-                    // objects uploaded before bucket suspended
-                    targetCount = 'masterCount';
-                    targetData = 'masterData';
-                } else {
-                    // null version
-                    // include current objects in nonversioned bucket and
-                    // objects uploaded after bucket suspended
-                    targetCount = 'nullCount';
-                    targetData = 'nullData';
-                }
-                Object.keys(data).forEach(metricLevel => {
-                    // metricLevel can only be 'bucket', 'location' or 'account'
-                    if (validStorageMetricLevels.has(metricLevel)) {
-                        Object.keys(data[metricLevel]).forEach(resourceName => {
-                            // resourceName can be the name of bucket, location or account
-                            if (!collRes[metricLevel][resourceName]) {
-                                collRes[metricLevel][resourceName] = {
-                                    masterCount: 0,
-                                    masterData: 0,
-                                    nullCount: 0,
-                                    nullData: 0,
-                                    versionCount: 0,
-                                    versionData: 0,
-                                    deleteMarkerCount: 0,
-                                };
-                            }
-                            collRes[metricLevel][resourceName][targetData] += data[metricLevel][resourceName];
-                            collRes[metricLevel][resourceName][targetCount]++;
-                            collRes[metricLevel][resourceName].deleteMarkerCount += res.value.isDeleteMarker ? 1 : 0;
+        while (await cursor.hasNext()) {
+            const res = await cursor.next();
+            const bucketCreationDate = await (new Promise((resolve, reject) => {
```

Comment:
do we need to compute the bucketCreationDate _inside_ the loop: the loop iterates over a single bucket, so it should be enough to read the bucketCreationDate once before going through the cursor ?

e.g. similar to how we get the bucketInfo once at the beginning (actually, as parameter) of this function

Reply:
changed to read the usersbucket once to get all buckets' creationDates
https://github.com/scality/s3utils/pull/286/commits/0f66c93a8e3950a55c845009f53ec4ee2311832d